commit 46934f682effe951f12bdf6f9ac71aea78d380e9
Author: Casey Connolly <kcxt@postmarketos.org>
Date:   Fri Oct 3 15:09:49 2025 +0200

    board/qualcomm: qcom-phone.env: set up userdata subpartitions with blkmap
    
    Use blkmap to expose the subpartitions on userdata which most current
    installs use, this allows us to boot from the ESP in the subpartition.
    
    Signed-off-by: Casey Connolly <kcxt@postmarketos.org>

diff --git a/board/qualcomm/qcom-phone.env b/board/qualcomm/qcom-phone.env
index 56603c8dacc..fac2ddde758 100644
--- a/board/qualcomm/qcom-phone.env
+++ b/board/qualcomm/qcom-phone.env
@@ -8,8 +8,8 @@ stderr=serial,vidconsole
 # allocate its buffer at runtime.
 fastboot=fastboot -l $fastboot_addr_r usb 0
 
-# Always probe for UFS storage, though it should be done by board code.
-preboot=scsi scan
+# Set up a block map pointing to our subpartition
+preboot=scsi scan; part start scsi 0 userdata userdata_start; part size scsi 0 userdata userdata_size; blkmap create root; blkmap map root 0 0x${userdata_size} linear scsi 0 0x${userdata_start}
 
 # Shortcut to enable USB serial gadget and disable bootretry
 serial_gadget=setenv stdin serial,button-kbd,usbacm; setenv stdout serial,vidconsole,usbacm; setenv stderr serial,vidconsole,usbacm; setenv bootretry -1; echo Enabled U-Boot console serial gadget
